spec/models/miq_request_task/state_machine_spec.rb

```diff
@@ -1,15 +1,19 @@
 RSpec.describe MiqRequestTask do
+  let(:task) { FactoryBot.create(:miq_request_task) }
+
   context "::StateMachine" do
     context "#signal" do
       it "will deal with exceptions" do
-        task = FactoryBot.create(:miq_request_task)
         allow(task).to receive_messages(:miq_request => double("MiqRequest").as_null_object)
         exception = String.xxx rescue $!
-        allow(task).to receive(:some_state).and_raise(exception)
+        allow(task).to receive(:send).with(:some_state).and_raise(exception)
```

Comment:
Wish this were calling `call_original` or something

I wonder if all these `expect` calls are stubbing out `miq_request_task` and so that is why the `task.status` is not properly getting set

Reply:
it seems calling `some_state` would raise an error anyway.
Wonder why we need to `allow` it if it will already raise an exception

Reply:
@kbrock I think that was done to test the logging.